_pytest/compat.py

```diff
@@ -237,5 +237,7 @@ def safe_str(v):
         try:
             return str(v)
         except UnicodeError:
+            if not isinstance(v, unicode):
+                v = unicode(v)
             errors = 'replace'
             return v.encode('ascii', errors)
```

Comment:
Seems good for me, the only thing is that I find it a bit weird that safe_str looses information if it has non-ascii chars (i.e.: usually the v.encode() would be something as v.encode('utf-8', 'replace') or v.encode(sys.getdefaultencoding(), 'replace'))

Anyways, this isn't related to this patch at all, so, +1 from the current patch from me ;)